webmappsrl/pap
# test(pap_home): add e2e tests for home buttons and user authentication

...

test(pap_info): add e2e tests

...

test(pap_waste_center_collection): add e2e tests

test(pap_home_servizi): add e2e tests

test(pap_calendar_page): add e2e tests

...

...

test(pap_reports): add e2e tests

...

refactor(e2e_tests): collected the useful functions for testing in a utils file

...

...

test(pap_reports_detail): add e2e tests

test(pap_trash_book): add e2e tests

test(pap_trash_book_detail): add e2e tests

test(pap_trash_book_type): add e2e tests

...

...

...

test(pap_ticket_reservation): add e2e tests

...

...

test(pap_abandonment_ticket): add e2e tests

...

test(pap_report_ticket): add e2e tests

This commit adds a new file, `pap-report-ticket.cy.ts`, which contains the implementation of the e2e tests for the "pap-report-ticket" feature. The file includes imports from various modules and defines constants and functions used in the tests.

The tests are organized into different describe blocks, each focusing on a specific step of the form. The tests verify the correct behavior of the form at each step, including opening action sheets, displaying correct labels and buttons, validating input fields, and navigating between steps.

The commit also includes changes to `test-utils.ts` to add a new utility function `getApiDateRange()` that generates start and stop date ranges for API calls based on the current date.

Finally, there is a minor change in `report-ticket.component.ts` where an import statement is added for `ReportTicketTypes` from `form.model`.

test(pap_info_ticket): add e2e tests

This commit add the code for the pap-info-ticket and update pap-abandonment-ticket forms. The changes include:
- Removing unused imports and variables
- Updating button labels and actions
- Refactoring form steps and their behaviors

These changes ensure that the forms function correctly and provide a better user experience.

test(pap_settings): add e2e tests

This commit adds the `pap-settings.cy.ts` file, which contains tests for the "GENERALI", "PASSWORD", "INDIRIZZI", and "AGGIUNGI INDIRIZZO" tabs in the pap-settings feature. It also includes the `test-utils.ts` file, which provides a utility function for logging in during e2e tests.

The `pap-settings.cy.ts` file contains tests to verify that the form fields in the "GENERALI" tab match the corresponding data from the API response. It also includes tests to check if the password fields are disabled/enabled correctly in the "PASSWORD" tab and verifies that zones and user addresses are displayed correctly in the "INDIRIZZI" tab. Additionally, it includes tests for adding a new address using the location modal.

The `test-utils.ts` file provides a utility function called `e2eLogin()` that performs login actions during e2e tests.

Co-authored-by: AI Assistant <assistant@example.com>

test(pap_sign_up): add e2e tests

- Refactor code to use arrow functions instead of function expressions
- Update test descriptions for clarity and consistency
- Fix typo in comment
- Add new file `pap-sign-up.cy.ts` with sign-up form tests
- Add new test case to verify the correct behavior of the form at each step
- Implement random click on the map to select an address and verify it using Nominatim API
- Verify that the selected user type label matches one of the available user types from API response
- Enable "next" button only when all required fields are filled in each step
